src/target/move.c

```diff
@@ -119,19 +126,48 @@ void disable_walls_control(void)
 void target_straight(int32_t start, float distance, float speed)
 {
 	int32_t target_distance;
-	uint32_t target_ticks;
 
 	set_target_angular_speed(0.);
-	target_distance = start + (int32_t)(distance * MICROMETERS_PER_METER) -
-			  (int32_t)required_micrometers_to_speed(speed);
-	set_target_linear_speed(max_linear_speed);
-	target_ticks = required_ticks_to_speed(speed);
-	while (get_encoder_average_micrometers() < target_distance)
-		;
-	set_target_linear_speed(speed);
-	target_ticks += get_clock_ticks();
-	while (get_clock_ticks() < target_ticks)
-		;
+
+	target_distance = start + (int32_t)(distance * MICROMETERS_PER_METER);
+	if (distance > 0) {
+		set_target_linear_speed(max_linear_speed);
+		while (get_encoder_average_micrometers() <
+		       target_distance -
+			   (int32_t)required_micrometers_to_speed(speed))
+			;
+	} else {
+		set_target_linear_speed(-max_linear_speed);
+		while (get_encoder_average_micrometers() >
+		       target_distance +
+			   (int32_t)required_micrometers_to_speed(speed))
+			;
+	}
+	change_speed(speed);
```

Comment:
It would be clear to do it without change_speed function

Reply:
Removed. :+1:

Reply:
5a974fda1f47e6f16b6bfb0c13487e17f4060624